nanovdb/nanovdb/PNanoVDB.h

```diff
@@ -2575,16 +2577,78 @@ PNANOVDB_FORCE_INLINE pnanovdb_bool_t pnanovdb_hdda_zero_crossing(
         ijk = pnanovdb_hdda_pos_to_ijk(PNANOVDB_REF(pos_start));
         dim = pnanovdb_uint32_as_int32(pnanovdb_readaccessor_get_dim(PNANOVDB_GRID_TYPE_FLOAT, buf, acc, PNANOVDB_REF(ijk)));
         pnanovdb_hdda_update(PNANOVDB_REF(hdda), origin, direction, dim);
+        // skip over tile
         if (hdda.dim > 1 || !pnanovdb_readaccessor_is_active(grid_type, buf, acc, PNANOVDB_REF(ijk)))
         {
             continue;
         }
+        // only check active values
         while (pnanovdb_hdda_step(PNANOVDB_REF(hdda)) && pnanovdb_readaccessor_is_active(grid_type, buf, acc, PNANOVDB_REF(hdda.voxel)))
         {
             ijk = hdda.voxel;
             pnanovdb_address_t address = pnanovdb_readaccessor_get_value_address(PNANOVDB_GRID_TYPE_FLOAT, buf, acc, PNANOVDB_REF(ijk));
-            PNANOVDB_DEREF(v) = pnanovdb_read_float(buf, address);
-            if (PNANOVDB_DEREF(v) * v0 < 0.f)
+            PNANOVDB_DEREF(valueAtHit) = pnanovdb_read_float(buf, address);
+            // checks if the sign has changed
+            if (PNANOVDB_DEREF(valueAtHit) * v0 < 0.f)
+            {
+                PNANOVDB_DEREF(thit) = hdda.tmin;
+                return PNANOVDB_TRUE;
+            }
+        }
+    }
+    return PNANOVDB_FALSE;
+}
+
+PNANOVDB_FORCE_INLINE pnanovdb_bool_t pnanovdb_hdda_next_hit(
+    pnanovdb_grid_type_t grid_type,
+    pnanovdb_buf_t buf,
+    PNANOVDB_INOUT(pnanovdb_readaccessor_t) acc,
+    PNANOVDB_IN(pnanovdb_vec3_t) origin, float tmin,
+    PNANOVDB_IN(pnanovdb_vec3_t) direction, float tmax,
+    PNANOVDB_INOUT(float) thit,
+    PNANOVDB_INOUT(float) valueAtHit,
+    PNANOVDB_INOUT(float) minimalValue

```

Comment:
That probably only needs to be PNANOVDB_IN(float) minimalValue

Reply:
Hello Ken, I agree with you and I removed the minimalValue. I also renamed the function to be the same of NanoVDB.